### DISCUSSION TITLE
Nvidia Flare: ML/DL models Limitations and Potentials

### DISCUSSION THREAD

**Discussion by Datumologist** [NVIDIA/NVFlare#1726] [Q&A] [answered]
I am interested in understanding the limitations of Nvidia Flare. I’m aware that it supports several frameworks, such as PyTorch, TensorFlow, Numpy, or MONAI. I am wondering if any model built using these frameworks would be able to function in a federated manner. Is there a particular subset of models that would be better suited for use in this platform?

Moreover, I relatively sure that I can construct models such as neural networks, logistic regression, linear regression, support vector machines, k-means, XGBoost, and random forests using Nvidia Flare. However, I am curious about the feasibility of implementing other models, such as Naive Bayes, decision trees or gradient boosting, just for mentioning some. Would these models also be compatible with Nvidia Flare?

Thanks in advance!

**Comment by YuanTingHsieh** [comment]
Hi @Datumologist  thanks for the questions!

My responses:

1. Yes, any models should be able to function in a federated manner. As for the NVFlare limitation, currently we DO NOT support models that are extremely BIG (say like 80 GB), we are working on that and will release a enhance version to solve that in next release.
2. Is there a particular subset of models that would be better suited for use in this platform?
    => I don't think NVFlare prefer any models over another subset of models, you should be able to experience a wide variety of models.
3. For each client to run Naive Bayes, decision trees or any type of gradient boosting is surely working. NVFlare does not restrict the type of applications. The real problem I think is HOW to combine these locally trained models in a federated manner, this part is a active research area. Tagging @ZiyueXu77 and @holgerroth to comment more, thanks in advance.

**Reply by ZiyueXu77** [reply, marked as answer]
@YuanTingHsieh gives a detailed answer, adding an additional point for "compatibility":
XGBoost and random forest are two of the most widely used methods in "decision trees or gradient boosting" family, as for Naive Bayes, it can be formulated in a distributed/federated way (and implemented with NVFlare) by collecting the stats. So they can all be implemented with NVFlare
As yuanting mentioned, the real question is "would/how these models be compatible with **federated learning**", and there is no "compatibility" issue on NVFlare side as long as we can formulated a method under a federated setting, and have sufficient API support from the package we use (e.g. scikit learn).
We will have a blog post to further explain this.
